File: go-client-sdk/pkg/models/operations/getbatchcrmeventattendee.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetBatchCrmEventAttendeeQueryParams struct {
	AccessToken string  `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool   `queryParam:"style=form,explode=true,name=allFields"`
	Ids         *string `queryParam:"style=form,explode=true,name=ids"`
}

type GetBatchCrmEventAttendeeResponseBody struct {
	EventAttendees []shared.EventAttendee `json:"eventAttendees,omitempty"`
	InvalidIds     []string               `json:"invalidIds,omitempty"`
}

type GetBatchCrmEventAttendeeRequest struct {
	QueryParams GetBatchCrmEventAttendeeQueryParams
}

type GetBatchCrmEventAttendeeResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetBatchCrmEventAttendeeResponseBody
}

File: typescript-client-sdk/src/sdk/models/operations/getbatchcrmeventattendee.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetBatchCrmEventAttendeeQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=ids" })
  ids?: string;
}


export class GetBatchCrmEventAttendeeResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=eventAttendees", elemType: shared.EventAttendee })
  eventAttendees?: shared.EventAttendee[];

  @SpeakeasyMetadata({ data: "json, name=invalidIds" })
  invalidIds?: string[];
}


export class GetBatchCrmEventAttendeeRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetBatchCrmEventAttendeeQueryParams;
}


export class GetBatchCrmEventAttendeeResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetBatchCrmEventAttendeeResponseBody;
}
